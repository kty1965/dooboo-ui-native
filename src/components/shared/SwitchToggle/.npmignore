tsconfig.tsbuildinfo
index.tsx
tsconfig.json
