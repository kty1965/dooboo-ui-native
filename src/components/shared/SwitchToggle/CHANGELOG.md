[0.3.1]

- Fix `react-native` import so that it can correctly import in project.
