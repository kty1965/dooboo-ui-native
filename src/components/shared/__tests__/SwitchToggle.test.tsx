import * as React from 'react';

// Note: test renderer must be required after react-native.
import SwitchToggle from '../SwitchToggle';
import { TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';

const component = (props?): React.ReactElement => {
  return <SwitchToggle {...props} />;
};

describe('[SwitchToggle]', (): void => {
  it('should render without crashing', (): void => {
    const rendered = renderer.create(component());
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  describe('[SwitchToggle] Interaction', (): void => {
    it('should simulate onPress', (): void => {
      const handlePress = jest.fn();
      const rendered = renderer.create(
        component({
          testID: 'SWITCH_ID',
          onPress: handlePress,
        }),
      );
      const switchToggle = rendered.root.findByType(TouchableOpacity);
      renderer.act(() => {
        switchToggle.props.onPress();
      });
      expect(handlePress).toHaveBeenCalled();
    });

    it('should toggle switchOn on press', () => {
      const props = {
        switchOn: false,
      };
      const rendered = renderer.create(component(props));
      const switchToggle = rendered.root.findByType(TouchableOpacity);

      renderer.act(() => {
        switchToggle.props.onPress();
      });
      expect(props.switchOn).toBeFalsy();

      // renderer.act(() => {
      //   switchToggle.props.onPress();
      // });
      // expect(props.switchOn).toBeTruthy();

      renderer.act(() => {
        switchToggle.props.onPress();
      });
      expect(props.switchOn).toBeFalsy();
    });
  });
});
