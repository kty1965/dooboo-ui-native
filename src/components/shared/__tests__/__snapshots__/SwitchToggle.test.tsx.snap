// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SwitchToggle] should render without crashing 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(215, 215, 215, 1)",
        "borderRadius": 18,
        "flexDirection": "row",
        "height": 36,
        "padding": 3,
        "width": 72,
      }
    }
  >
    <View>
      <Text />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 15,
          "height": 30,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "width": 30,
        }
      }
    >
      <View>
        <Text />
      </View>
    </View>
    <View>
      <Text />
    </View>
  </View>
</View>
`;
